src/app/components/game-components/abstract-game-component/AbstractGameComponent.ts

```diff
@@ -20,10 +19,9 @@ import { TutorialStep } from '../../wrapper-components/tutorial-game-wrapper/Tut
     styleUrls: ['./abstract-game-component.css'],
 })
 export abstract class AbstractGameComponent<M extends Move,
-                                            S extends GamePartSlice,
+                                            S extends GameState<unknown, unknown>,
```

Comment:
y'a pas moyen de spécifier les arguments de type ? Quels sont ils ?

Reply:
ah, d'après ce que je vois c'est un Coord et un Piece, donc soit on reste sur du unknown, unknown, soit sur C et P qui sont aussi des arguments de type de `AbstractGameComponent`, ce qui a aussi du sens et est peut-être plus compréhensible

Reply:
C et P ne sont pas encore des arguments d'*AbstractGameComponent*, on peut les y rajouter ceci dit mais un GameCompo en a rien à faire je crois de savoir ça.

En revanche, j'envisage de rajouter Rules dans les argument d'*AbstractGameComponent* ce qui permettrait d'instancienr this.rules dans le constructeur plus esthétiquement pour les deux trois compo qui utilisent pas juste rules.choose et rules.node ! T'en penses quoi ?

Reply:
en effet il en a rien à faire, du coup ptet préférer un alias de GameState<unknown, unknown> pour éviter la répétition ?

Ça me semble mieux pour Rules oui !

Reply:
alias ajouté!